### PR TITLE
Move setup.py and version into caffe2 folder.

### PR DESCRIPTION
**Pull request by orionr**
* Prep for repository movement.

### CODE REVIEW

**Review by pjh5 — APPROVED** [review]

File: caffe2/setup.py [hunk 1]
```diff
@@ -20,8 +20,8 @@
 import sys
 from textwrap import dedent
 
-TOP_DIR = os.path.realpath(os.path.dirname(__file__))
-SRC_DIR = os.path.join(TOP_DIR, 'caffe2')
+SRC_DIR = os.path.realpath(os.path.dirname(__file__))
+TOP_DIR = os.path.join(SRC_DIR, '..')
```
Comment (bddppq) [inline caffe2/setup.py]: 1. '..' is not truly cross-platform, there `os.pardir`
2. We should resolve the realpath instead of leaving `..` in it.

**Review by bddppq — CHANGES_REQUESTED** [review]
`find_packages` should also need changes

**Comment by bddppq** [comment]
To test whether this will work, you should create a PR in onnx-fb-universe repo with `install.sh` modified and `repos/caffe2` points to this caffe2 PR's commit.

**Comment by orionr** [comment]
Good catch on `find_packages()` - changing to `setuptools.find_packages(TOP_DIR)`

**Comment by orionr** [comment]
Thanks @bddppq - universe PR up at https://github.com/onnxbot/onnx-fb-universe/pull/1358

**Review by pjh5 — APPROVED** [review]

**Comment by ezyang** [comment]
Comments should call out that using absolute paths is important (relative paths don't work.) Does this work with `python setup.py develop`?

**Comment by orionr** [comment]
We decided to rename instead - closing.

**Pull request closed without merging**